Bottom: 6a4f2cab8af33eb85c43d648a1196e638eef4668
Top:    a2f313c516873e9295c2b25e6a6d09149f455644
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 20:11:41 +0000

Workaround crash where contour is passed as NULL to poly_PreContour












---

diff --git a/src/polygon.c b/src/polygon.c
index e28f584..dfcf4b6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
